### DISCUSSION TITLE
Layer2模式下，openelb-manager是如何选择代理节点的？

### DISCUSSION THREAD

**Discussion by bluven** [openelb/openelb#342] [Q&A]
根据这个文档https://openelb.io/docs/concepts/layer-2-mode/ 里的描述，代理节点（代理服务流量的节点，我没看到你们对这个节点的定义，暂时这么称呼吧）是被随机选出来的
```
OpenELB installed in the Kubernetes cluster randomly selects a node (worker 1 in this example) to handle Service requests. After that, OpenELB sends an ARP/NDP packet to the router, which maps the Service IP address to the MAC address of worker 1.
```
有一些疑问：
1. 这个代理节点是每一个服务随机选一个还是所有的loadbalancer 服务都选同一个节点？
2. 我做实验的过程中，想让另一个节点也被选中，但反复重启openelb-manager都依然会选择之前选中的节点，有什么法子影响选择过程吗？

**Comment by renyunkang** [comment]
随机选取节点，无法干预选择的过程，具体的你可以参照这里：https://github.com/openelb/openelb/blob/master/pkg/controllers/lb/controller.go#L211C1-L241C54

**Reply by bluven** [reply]
我的集群环境是边缘计算场景，现在用Kubeedge + fabedge, 选了openELB做实验，目前遇到的问题是openelb选择代理节点的时候是随机的，不管是BGP，layer2还是vip，都会把边缘节点作为可能的代理节点，但实际上边缘节点不适合作为代理节点，建议openELB加上节点筛选功能。
